### PR TITLE
TRUNK-5922 Switching from Hibernate Mappings to Annotations for DrugR…

### PR DESCRIPTION
**Pull request by UNCANNY69**
Switching from Hibernate Mappings to Annotations for DrugReferenceMap Domain.

## Description of what I changed

Switched from Hibernate Mappings to Annotations

## Issue I worked on

see https://openmrs.atlassian.net/browse/TRUNK-5922

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by mherman22** [comment]
Hey @UNCANNY69 , have you taken a look at the [OpenMRS pull request tips](https://openmrs.atlassian.net/wiki/spaces/docs/pages/25477199/Pull+Request+Tips)?

**Comment by UNCANNY69** [comment]
Hi @mherman22 , thanks for pointing it out I have read tips and figured out I have pointed the merge to wrong branch. Rest of the mentioned tips have been followed.

File: api/src/main/java/org/openmrs/DrugReferenceMap.java [hunk 1]
```diff
@@ -12,45 +12,74 @@
...
+import org.hibernate.annotations.Cascade;
+import org.hibernate.annotations.Parameter;
+import org.hibernate.annotations.GenericGenerator;
 import org.hibernate.envers.Audited;
 import org.hibernate.search.annotations.ContainedIn;
 import org.hibernate.search.annotations.DocumentId;
 import org.hibernate.search.annotations.IndexedEmbedded;
+import org.hibernate.annotations.CascadeType;
+
+import javax.persistence.*;
```
Comment (ManojLL) [inline api/src/main/java/org/openmrs/DrugReferenceMap.java:24]: please set up your IDE with openmrs, https://openmrs.atlassian.net/wiki/spaces/Archives/pages/25506949/How-To+Setup+And+Use+Your+IDE

we do not use wildcard imports

Comment (UNCANNY69) [reply api/src/main/java/org/openmrs/DrugReferenceMap.java]: I finished setting up the IDE according to the documentation and resolved the above mentioned wildcard import.

**Comment by ManojLL** [comment]
@UNCANNY69 your branch name should be  issue ID (eg: TRUNK-4567)

**Review by ManojLL — CHANGES_REQUESTED** [review]

**Pull request closed without merging**